### DISCUSSION TITLE
Generic or detect new workload controller types in Linkerd

### DISCUSSION THREAD

**Discussion by jonasdkhansen** [linkerd/linkerd2#4893] [General]
I'm using Argo Rollouts for doing automatic blue/green and canary deployments. Argo Rollouts creates a new custom resource definition called Rollouts, which is a clone of Deployments, but with some extra config. So to use this, you have to change your deployments to rollouts, and now this type is nowhere to find in Linkerd, because the controller is not known.

My suggestion is to automatic detect new workload controller types, or give the ability to add them to Linkerd.

**Comment by ihcsim** [comment]
I am actually surprised that to use Argo Rollouts, one will have to replace all their `Deployment` resources with the `Rollout` custom resource. Are there no ways to define labels matching in the `Rollout` custom resource to select existing `Deployment` resources?
